src/app/compiler/compiler-utils.js

```diff
@@ -0,0 +1,43 @@
+const semver = require('semver')
+/* global Worker */
+
+/**
+ * Checks if the worker can be used to load a compiler.
+ * checks a compiler whitelist, browser support and OS.
+ */
+export function canUseWorker (selectedVersion) {
+  // Following restrictions should be deleted when Solidity will release fixed versions of compilers.
+  // See https://github.com/ethereum/remix-ide/issues/2461
+  const isChrome = !!window.chrome
+  const os = retrieveOS()
+  // define a whitelist for Linux
+  const linuxWL = ['0.4.26', '0.5.3', '0.5.4', '0.5.5']
+  const version = semver.coerce(selectedVersion)
+  // defining whitelist for chrome
+  let isFromWhiteList = false
+  switch (os) {
```

Comment:
What about macOS, Android, iOS ?

Reply:
Rob said macOS is fine. why should we check for android?